benchmark/librbdfio.py

```diff
@@ -60,41 +70,120 @@ def __init__(self, archive_dir, cluster, config):
 
+
+    # Copy the FIO global options into a dictionary
+    def set_global_fio_options(self):
+        fio_opts = {}
+        fio_opts['time_based'] = self.time_based
+        fio_opts['ramp'] = self.ramp
+        fio_opts['iodepth'] = self.iodepth
+        fio_opts['numjobs'] = self.numjobs
+        fio_opts['mode'] = self.mode
+        fio_opts['end_fsync'] = self.end_fsync
+        fio_opts['rwmixread'] = self.rwmixread
+        fio_opts['rwmixwrite'] = self.rwmixwrite
+        fio_opts['log_avg_msec'] = self.log_avg_msec
+        fio_opts['op_size'] = self.op_size
+
+        self.global_fio_options = fio_opts
+
+
+    # Restore the global values that are set before each workload
+    def restore_global_fio_options(self):
+        self.time_based = self.global_fio_options['time_based']
+        self.ramp = self.global_fio_options['ramp']
+        self.iodepth = self.global_fio_options['iodepth']
+        self.numjobs = self.global_fio_options['numjobs']
+        self.mode = self.global_fio_options['mode']
+        self.end_fsync = self.global_fio_options['end_fsync']
+        self.rwmixread = self.global_fio_options['rwmixread']
+        self.rwmixwrite = self.global_fio_options['rwmixwrite']
+        self.log_avg_msec = self.global_fio_options['log_avg_msec']
+        self.op_size = self.global_fio_options['op_size']
+
+
+    def get_workloads(self):
+        if self.workloads:
+            logger.info(" %d Workloads:\n    %s", len(self.workloads.keys()),
+                        #', '.join([pprint.pformat(x) for x in self.workloads]).replace("\n", "\n    "))
+                        pprint.pformat(self.workloads).replace("\n", "\n    "))
+
+
+    def get_precondition(self):
+        logger.info("Precond:\n    %s",
+                    pprint.pformat(self.precondition).replace("\n", "\n    "))
+
+
     def exists(self):
         if os.path.exists(self.out_dir):
             logger.info('Skipping existing test in %s.', self.out_dir)
             return True
         return False
 
+
     def initialize(self):
         super(LibrbdFio, self).initialize()
 
         # Clean and Create the run directory
         common.clean_remote_dir(self.run_dir)
         common.make_remote_dir(self.run_dir)
 
-        logger.info('Pausing for 60s for idle monitoring.')
-        monitoring.start("%s/idle_monitoring" % self.run_dir)
-        time.sleep(60)
-        monitoring.stop()
+        if not self.skip_monitoring:
+            logger.info('Pausing for 60s for idle monitoring.')
+            monitoring.start("%s/idle_monitoring" % self.run_dir)
+            time.sleep(5)
+            monitoring.stop()
 
         common.sync_files('%s/*' % self.run_dir, self.out_dir)
 
         # Create the recovery image based on test type requested
         if 'recovery_test' in self.cluster.config and self.recov_test_type == 'background':
             self.mkrecovimage()
+
+        self.get_workloads()
+        self.get_precondition()
```

Comment:
Following on from one of my previous comments, I'd suggest the following:

```suggestion
    def initialize(self):
       ...
       if (self.workloads):
         self.set_global_fio_options()
         logger.info(" %d Workloads:\n....)
         logger.info("Precond:\n ...)
      ...
```

Reply:
I've refactored in an equivalent way, only the precondition has now been removed since was not used, probably a feature for a next PR